lxd/instance/drivers/driver_qemu.go

```diff
@@ -6391,6 +6391,82 @@ func (d *qemu) Info() instance.Info {
 	return data
 }
 
+// checkQMPCommand runs QEMU and checks if the given QMP command is available.
```

Comment:
This duplicates most of `checkFeature()` and causes LXD to start another QEMU process (the 3rd) during start up in order to probe QEMU features.

Instead it would be preferable to combine this function into `checkFeature()` and modify its signature to be `checkFeature(qemuPath) ([]string, error)` where a non-nil error indicates the QEMU process started OK (VMs supported), and the slice contains any optional features supported.

Then, as we actually add disks by using `m.AddBlockDevice()` via QMP these days, we should really use QMP inside `checkFeature()` to check for io_uring support.

This would then allow us to start a single QEMU process and test for both io_uring support and the CPU hotplug support via QMP, and it keeps all of the feature checking together.

What do you think?

Reply:
> where a non-nil error indicates the QEMU process started OK (VMs supported), and the slice contains any optional features supported.

I believe you mean a nil-error indicates VM support?

> What do you think?

Sounds good.

Reply:
Ah yes indeed :)